lib/keycard/proxied_resolver.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Keycard
+  # This resolver should be used when the application will be served behind a
+  # reverse proxy. It relies on the trusted relationship with the proxy to use
+  # HTTP headers for forwarded values.
+  class ProxiedResolver
+    # The typical headers forwarded are X-Forwarded-User and X-Forwarded-For,
+    # which, somewhat confusingly, are transposed into HTTP_X_REMOTE_USER and
+    # HTTP_X_FORWARDED_FOR once the Rack request is assembled.
+    def attributes_for(request)
+      {
+        username: request.env['HTTP_X_REMOTE_USER'],
+        client_ip: request.env['HTTP_X_FORWARDED_FOR']
```

Comment:
We should be able to get these via `request.get_header('X-Remote-User')` and `request.get_header('X-Forwarded-For')`.

Reply:
Also- how would Checkpoint handle the very typical case of receiving e.g.

```ruby
{ 
  username: nil,
  client_ip: '10.0.0.1'
}```?

Reply:
What cases do you envision the application using the IP address from keycard? Should this return an `IPAddr` instead of a string for the IP address?

Reply:
It turns out that the headers are always prepended with HTTP and capitalized, and that `get_header` is not really about headers at all. It's just a shortcut for `.env[...]`. I think you're right about not really needing the client IP exposed, at least for now... I think you'll like the updated solution I have ready.